src/Workspaces/Remote/Core/Services/AssetStorage.cs

```diff
@@ -75,9 +75,9 @@ public AssetStorage(TimeSpan cleanupInterval, TimeSpan purgeAfter, TimeSpan gcAf
             Task.Run(CleanAssetsAsync, CancellationToken.None);
         }
 
-        public AssetSource? AssetSource => _assetSource;
+        public IAssetSource? AssetSource => _assetSource;
 
-        public void SetAssetSource(AssetSource assetSource)
+        public void SetAssetSource(IAssetSource assetSource)
             => _assetSource = assetSource;
```

Comment:
should this throw if we've already set the asset source before?  i don't really understand the lifetimes/ownership here.

Reply:
Yes. Will add throw.